controllers/openstackbackuprequest_controller.go

```diff
@@ -363,16 +363,22 @@ func (r *OpenStackBackupRequestReconciler) ensureLoadBackup(
 	backup *ospdirectorv1beta2.OpenStackBackup,
 ) error {
 	// Create all CRs in the spec first, and set their status to some initial state
+	// NOTE: For some reason, the "status" of the various network-related CRs is being
+	//       removed after the CR is applied.  To work around this, we create a deep
+	//       copy of the status to apply after the CR's spec has been applied.
 
 	msg := fmt.Sprintf("OpenStackBackup %s initial load", backup.Name)
 
 	// OpenStackNets
 	for _, item := range backup.Spec.Crs.OpenStackNets.Items {
+		status := item.Status.DeepCopy()
```

Comment:
Looking at the code it seems that we use CreateOrPatch to restore each object. I wonder if that just doesn't set the status correctly. Also, had a question if on restoration if we should be calling a controllerutil.SetControllerReference. Perhaps I missed this being done elsewhere though